src/main/java/com/google/sps/servlets/CreateUserServlet.java

```diff
@@ -0,0 +1,64 @@
+package com.google.sps.servlets;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.NullValue;
+import com.google.sps.data.User;
+
+import java.util.UUID;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/create-user")
+public class CreateUserServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Get the value entered in the form.
+    String username = request.getParameter("username").trim();
+    String userID = UUID.randomUUID().toString();
+
+    if (username == null || username == "") {
+        response.sendError(400, "empty username");
+    }
+
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+
+    Query<Entity> query =
```

Comment:
this runs in O(number of users), we could change this to O(1) by using userN as the datastore key then doing a datastore get